mods/gen2/scripts.js

```diff
@@ -117,6 +117,76 @@ exports.BattleScripts = {
 		this.singleEvent('AfterMove', move, null, pokemon, target, move);
 		if (!move.selfSwitch && target.hp > 0) this.runEvent('AfterMoveSelf', pokemon, target, move);
 	},
+	tryMoveHit: function (target, pokemon, move, spreadHit) {
+		let positiveBoostTable = [1, 1.33, 1.66, 2, 2.33, 2.66, 3];
+		let negativeBoostTable = [1, 0.75, 0.6, 0.5, 0.43, 0.36, 0.33];
+		let doSelfDestruct = true;
+		let damage = 0;
+
+		// First, check if the Pokémon is immune to this move.
+		if (move.ignoreImmunity !== true && !move.ignoreImmunity[move.type] && !target.runImmunity(move.type, true)) {
+			return false;
+		}
+
+		if (move.selfdestruct) {
+			this.faint(pokemon, pokemon, move);
+		}
+
+		// Now, let's calculate the accuracy.
+		let accuracy = move.accuracy;
+		if (accuracy !== true) {
+			accuracy = Math.floor(accuracy * 255 / 100);
+			if (move.ohko) {
+				if (pokemon.level >= target.level) {
+					accuracy += (pokemon.level - target.level) * 2;
+					accuracy = Math.min(accuracy, 255);
+				} else {
+					this.add('-immune', target, '[ohko]');
+					return false;
+				}
+			}
+			if (!move.ignoreAccuracy) {
+				if (pokemon.boosts.accuracy > 0) {
+					accuracy *= positiveBoostTable[pokemon.boosts.accuracy];
+				} else {
+					accuracy *= negativeBoostTable[-pokemon.boosts.accuracy];
+				}
+			}
+			if (!move.ignoreEvasion) {
+				if (target.boosts.evasion > 0 && !move.ignorePositiveEvasion) {
+					accuracy *= negativeBoostTable[target.boosts.evasion];
+				} else if (target.boosts.evasion < 0) {
+					accuracy *= positiveBoostTable[-target.boosts.evasion];
+				}
+			}
+			accuracy = Math.min(Math.floor(accuracy), 255);
+			accuracy = Math.max(accuracy, 1);
+		}
+		accuracy = this.runEvent('ModifyAccuracy', target, pokemon, move, accuracy);
+		accuracy = Math.max(accuracy, 0);
```

Comment:
Math.max(true, 0) = 1, so moves that never miss will be missing 255 out of 256 times.